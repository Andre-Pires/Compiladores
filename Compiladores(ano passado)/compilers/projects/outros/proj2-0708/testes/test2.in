int even(int n){
  int m;
  if(n==0)
    return 1;
  else{
    m= n - 1;
    if(odd(n))
      return 1;
    else
      return 0;
  }
}
