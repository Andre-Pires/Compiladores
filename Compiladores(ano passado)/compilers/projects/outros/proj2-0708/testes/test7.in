int sum;
int a[10];
int b[10];

int  read();
void write(int c);

void init(){
  int i;

  i = 0;
  while(i < 10){
    a[i] = 0;
    b[i] = 0;
    i = i + 1;
  }
  sum=0;
}

int findMax(int idx){

}

void main(int argc, char *argv){
  int i;
  int m;
  int max;

  init();
  max = read();
  m = findMax(max);
}

