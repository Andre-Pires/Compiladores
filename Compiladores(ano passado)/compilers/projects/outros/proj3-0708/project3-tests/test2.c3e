init: 	i = 7
	t1 = i
	xy[t1] = 3
	t2 = i
	b = xy[t2]
	putparam b
  	call iwrite, 1
	halt
