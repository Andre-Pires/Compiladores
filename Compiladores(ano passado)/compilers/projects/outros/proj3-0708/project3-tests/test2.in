/* straight line code with scalars, arrays no input */
int xy[10];
int b;
int z;
void main(){
  int i;

  i = 7;
  xy[i] = 3;
  b = xy[i];
  iwrite(b);
}
