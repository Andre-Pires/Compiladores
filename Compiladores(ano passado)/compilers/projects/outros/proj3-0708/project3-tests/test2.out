C3E Simulator 07/10/07, 22:55:13
3
*** Symbol table dump ***
init = 0
i = 7
t1 = 7
xy = 0
t2 = 7
b = 3
