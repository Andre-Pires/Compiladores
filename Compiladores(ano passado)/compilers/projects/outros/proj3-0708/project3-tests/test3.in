/* straight line code with scalars, arrays no input */
int xy[10];
int b;
int z;
void main(){
  int i;
  int t1;

  i = 7;
  xy[i] = 3;
  b = xy[i];
  t1 = b + 1;
  iwrite(t1);
}
