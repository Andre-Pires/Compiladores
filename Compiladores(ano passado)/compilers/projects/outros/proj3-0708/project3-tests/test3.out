C3E Simulator 07/10/07, 22:55:13
4
*** Symbol table dump ***
init = 0
i = 7
t1 = 7
xy = 0
t2 = 7
b = 3
t3 = 4
