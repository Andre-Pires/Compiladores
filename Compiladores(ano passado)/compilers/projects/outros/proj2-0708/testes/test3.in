#include <test2.in>

int a[100];
int b[20];

void main(){
  int x; int y; int z;
  read(x); read(y);
  z= x + y;
  if(even(x)) print();
  else if(even(y)) print(y);
       else if(even(z)){print(z);}
}
