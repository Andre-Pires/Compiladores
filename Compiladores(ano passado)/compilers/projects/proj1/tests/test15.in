#include <test7.in>
#include <stdio.h>
#include <test3.in>

int x;
int s(int x){
  return x + 1;
}

#include <test2.in>
