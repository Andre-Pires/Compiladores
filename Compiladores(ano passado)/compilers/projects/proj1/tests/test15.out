1 271 int
1 257 x
1 59 ;
4 271 int
4 257 s
4 40 (
4 271 int
4 257 x
4 41 )
4 123 {
5 278 return
5 257 x
5 43 +
5 258 1
5 59 ;
5 125 }
3 271 int
3 257 f
3 40 (
3 271 int
3 257 x
3 41 )
3 123 {
4 271 int
4 257 y
4 59 ;
4 271 int
4 257 z
4 59 ;
5 275 if
5 40 (
5 257 x
5 265 <=
5 258 1
5 41 )
5 257 then
6 278 return
6 258 1
6 59 ;
7 276 else
7 123 {
7 257 y
7 61 =
7 257 x
7 45 -
7 258 1
7 59 ;
7 257 z
7 61 =
7 257 f
7 40 (
7 257 y
7 41 )
7 59 ;
7 278 return
7 257 x
7 42 *
7 257 z
7 59 ;
7 125 }
8 125 }
5 271 int
5 257 x
5 59 ;
6 271 int
6 257 s
6 40 (
6 271 int
6 257 x
6 41 )
6 123 {
7 278 return
7 257 x
7 43 +
7 258 1
7 59 ;
8 125 }
1 273 void
1 257 main
1 40 (
1 41 )
1 123 {
2 271 int
2 257 i
2 59 ;
2 271 int
2 257 j
2 59 ;
3 271 int
3 257 a
3 91 [
3 258 10
3 93 ]
3 59 ;
4 271 int
4 257 b
4 91 [
4 258 10
4 93 ]
4 59 ;
5 257 i
5 61 =
5 258 0
5 59 ;
6 257 j
6 61 =
6 258 10
6 59 ;
7 277 while
7 40 (
7 257 i
7 263 <
7 257 j
7 41 )
7 123 {
8 257 i
8 61 =
8 257 i
8 43 +
8 258 2
8 59 ;
8 257 j
8 61 =
8 257 j
8 43 +
8 258 1
8 59 ;
9 257 a
9 91 [
9 257 i
9 93 ]
9 61 =
9 257 b
9 91 [
9 257 j
9 93 ]
9 59 ;
10 125 }
11 125 }
#Total Number of Tokens Matched was 139
